### PR TITLE
Vendor older versions package-build & package-recipe for Emacs 24.*

### PR DESCRIPTION
**Pull request by immerrr**
This PR adds vendoring for earlier versions of `package-build` package that is compatible with Emacs 24.

### CODE REVIEW

File: cask-bootstrap.el [hunk 1]
```diff
@@ -47,25 +47,36 @@
...
+                 (require package))
+             (error
+              (unless package-archives
+                (add-to-list 'package-archives (cons "gnu" "https://elpa.gnu.org/packages/"))
+                (add-to-list 'package-archives (cons "melpa" "https://stable.melpa.org/packages/"))
+                (package-refresh-contents))
+              (message "installing package %S" package)
+              (package-install package)
+              (require package))))
+         cask-bootstrap-packages))
```
Comment (conao3) [inline cask-bootstrap.el:79]: Please revert this hunk to clear your change. If you want this change, please create another PR.

Comment (immerrr) [reply cask-bootstrap.el:79]: I am happy to revert the "messages" part, but the workflow where every package is checked and installed one-by-one is important. Previous behaviour was to install all the missing packages if one of the requires fails. Without this change cask will not find some other bootstrap package and still try to install `package-build` along with the others.

Comment (immerrr) [reply cask-bootstrap.el:79]: Ok, let me open a PR with just the change in logic and see if it works better.

Comment (conao3) [reply cask-bootstrap.el:79]: > I am happy to revert the "messages" part, but the workflow where every package is checked and installed one-by-one is important. Previous behaviour was to install all the missing packages if one of the requires fails. Without this change cask will not find some other bootstrap package and still try to install package-build along with the others.

Nice. It's reasonable.

Comment (conao3) [reply cask-bootstrap.el:76]: There are separated PR, #493

**Comment by conao3** [comment]
Could you rebase, please?

**Comment by immerrr** [comment]
Done!

**Comment by conao3** [comment]
Thanks! merge.

**Pull request merged**